src/Plugin/RetryPlugin.php

```diff
@@ -35,18 +45,30 @@
      * @param array $config {
      *
      *     @var int $retries Number of retries to attempt if an exception occurs before letting the exception bubble up.
+     *     @var callable $decider A callback that gets a request and an exception to decide if we should retry this or not.
+     *     @var callable $delay A callback to return how many milliseconds we should wait before trying again.
      * }
      */
     public function __construct(array $config = [])
     {
         $resolver = new OptionsResolver();
         $resolver->setDefaults([
             'retries' => 1,
+            'decider' => function (RequestInterface $request, Exception $e) {
+                return true;
+            },
+            'delay' => function (RequestInterface $request, Exception $e, $retries) {
+                return ((int) pow(2, $retries - 1)) * 1000;
```

Comment:
this could be a public static function on this class. then we could unit test it.